python/ray/state.py

```diff
@@ -385,6 +385,66 @@ def client_table(self):
 
         return _parse_client_table(self.redis_client)
 
+    def _job_table(self, job_id):
+        """Fetch and parse the job table information for a single job ID.
+
+         Args:
+            job_id: A job ID to get information about.
+         Returns:
+            A dictionary with information about the job ID in question.
```

Comment:
I think it would be better to define a `JobInfo` class instead of using a dictionary for this. It's only a few fields and no logic right now, but this will likely grow in scope over time. Would also improve documentation over just having the fields in a header comment for another function.

Reply:
Yeah I agree! Let's do this change in a followup PR, since all the other tables (tasks, objects, clients) are dictionaries in the same format at the moment and we should be consistent.

Other possible choices we could consider are json with json-schema (would make it easy to put this API behind a REST endpoint in the future), protobuf, namedtuple (meh), dataclass (meh bc might require new python), any thoughts? The cc @simon-mo

Reply:
Few considerations:
- there's a backport for dataclass `pip install dataclasses` so that won't be an issue. 
- in the api schema world, openapi schema is fairly big as well. (Kubernetes [use](https://kubernetes.io/docs/concepts/overview/kubernetes-api/) it)
- protobuf in python is hard to work with. Non-intuitive API, slow serialization
- dataclass > namedtuple